SimPEG/regularization/base.py

```diff
@@ -180,12 +352,290 @@ def deriv2(self, m, v=None):
+    :param numpy.ndarray weights: cell weights
+    :param bool reference_model_in_smooth: include the reference model in the
+        smoothness computation? (eg. look at Deriv of m (False) or Deriv of
+        (m-reference_model) (True))
+    :param numpy.ndarray weights: vector of cell weights (applied in all terms)
+    """
+
+    def __init__(
+        self, mesh, orientation="x", reference_model_in_smooth=False, **kwargs
+    ):
+        self.reference_model_in_smooth = reference_model_in_smooth
+
+        if orientation not in ["x", "y", "z"]:
+            raise ValueError("Orientation must be 'x', 'y' or 'z'")
+
+        if orientation == "y" and mesh.dim < 2:
+            raise ValueError(
+                "Mesh must have at least 2 dimensions to regularize along the "
+                "y-direction."
+            )
+        elif orientation == "z" and mesh.dim < 3:
+            raise ValueError(
+                "Mesh must have at least 3 dimensions to regularize along the "
+                "z-direction"
+            )
+        self._orientation = orientation
+
+        super().__init__(mesh=mesh, **kwargs)
+        self.set_weights(volume=self.regularization_mesh.vol)
+
+    @property
+    def _weights_shapes(self):
+        """Acceptable lengths for the weights
+
+        Returns
+        -------
+        tuple
+            A tuple of each acceptable lengths for the weights
+        """
+        n_active_f, n_active_c = getattr(
+            self.regularization_mesh, "aveCC2F{}".format(self.orientation)
+        ).shape
+        return [(n_active_f,), (n_active_c,)]
+
+    @property
+    def cell_gradient(self):
+        """Cell gradient operator"""
+        if getattr(self, "_cell_gradient", None) is None:
+            self._cell_gradient = getattr(
+                self.regularization_mesh, "cell_gradient_{}".format(self.orientation)
+            )
+        return self._cell_gradient
+
+    @property
+    def reference_model_in_smooth(self) -> bool:
+        """
+        Use the reference model in the model gradient penalties.
+        """
+        return self._reference_model_in_smooth
+
+    @reference_model_in_smooth.setter
+    def reference_model_in_smooth(self, value: bool):
+        if not isinstance(value, bool):
+            raise TypeError(
+                "'reference_model_in_smooth must be of type 'bool'. "
+                f"Value of type {type(value)} provided."
+            )
+        self._reference_model_in_smooth = value
+
+    def _delta_m(self, m):
+        if self.reference_model is None or not self.reference_model_in_smooth:
+            return m
+        return m - self.reference_model
+
+    @property
+    def _multiplier_pair(self):
+        return f"alpha_{self.orientation}"
+
+    def f_m(self, m):
+        """
+        Model gradient
+        """
+        dfm_dl = self.cell_gradient @ (self.mapping * self._delta_m(m))
+
+        if self.units == "radian":
```

Comment:
should we do a `.lower()` here? or I might have missed if this is unnecessary and done in the setter

Reply:
Yep, done.